cinema/views.py

```diff
@@ -56,3 +89,59 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action in ("retrieve", "list"):
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows")
+                        * F("cinema_hall__seats_in_row")
+                        - Count("tickets")
+                    )
+                )
+            )
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
```

Comment:
this can still work if the format is exactly as we expect, but normally this param should be of the date type (validated)